### PR TITLE
Wait4 sequence

### PR DESCRIPTION
**Pull request by tkrol88**

### CODE REVIEW

File: moler/event.py [hunk 1]
```diff
@@ -66,3 +66,7 @@ def get_long_desc(self):
 
     def get_short_desc(self):
         return "Event '{}.{}'".format(self.__class__.__module__, self)
+
+    def get_last_occurrence(self):
```
Comment (marcin-usielski) [inline moler/event.py]: what if self._occurred is None?

File: moler/event.py [hunk 2]
```diff
@@ -66,3 +66,7 @@ def get_long_desc(self):
...
+        if self._occurred is not None:
```
Comment (marcin-usielski) [inline moler/event.py]: from __init__: self._occurred = []. However list may be empty.

Comment (marcin-usielski) [inline moler/event.py]: Maybe check if command is done if not self.done() ?

File: moler/events/lineevent.py [hunk 3]
```diff
@@ -13,21 +13,19 @@
 
 
 class LineEvent(TextualEvent):
-    def __init__(self, detect_pattern=None, detect_patterns=list(), connection=None, till_occurs_times=-1):
+    def __init__(self, detect_patterns=list(), connection=None, till_occurs_times=-1, match='any'):
```
Comment (marcin-usielski) [inline moler/events/lineevent.py]: Whad do you think about  detect_patterns without default value?

Comment (marcin-usielski) [reply moler/events/lineevent.py]: Where is runner?

Comment (tkrol88) [reply moler/events/lineevent.py]: Good point. detect_patterns should be required parameter. I added runner for all events

File: moler/event.py [hunk 4]
```diff
@@ -66,3 +66,7 @@ def get_long_desc(self):
 
     def get_short_desc(self):
         return "Event '{}.{}'".format(self.__class__.__module__, self)
+
+    def get_last_occurrence(self):
+        if self._occurred:
+            return self._occurred[-1]
```
Comment (tkrol88) [inline moler/event.py]: return None

File: moler/events/lineevent.py [hunk 5]
```diff
@@ -13,21 +13,19 @@
...
 class LineEvent(TextualEvent):
-    def __init__(self, detect_pattern=None, detect_patterns=list(), connection=None, till_occurs_times=-1):
-        super(LineEvent, self).__init__(connection=connection, till_occurs_times=till_occurs_times)
-        self.detect_pattern = detect_pattern
+    def __init__(self, detect_patterns, connection=None, till_occurs_times=-1, match='any', runner=None):
+        super(LineEvent, self).__init__(connection=connection, runner=runner, till_occurs_times=till_occurs_times)
         self.detect_patterns = copy_list(detect_patterns)
         self.process_full_lines_only = False
+        self.match = match
+        self._prepare_parameters(match)
```
Comment (tkrol88) [inline moler/events/lineevent.py]: remove match

File: moler/events/lineevent.py [hunk 6]
```diff
@@ -66,6 +58,70 @@ def get_long_desc(self):
...
 
+    def get_parser(self, match):
+        parsers = {
+            "any": self._catch_any,
+            "all": self._catch_all,
+            "sequence": self._catch_sequence,
+        }
+        if match in parsers:
+            return parsers[match]
+
```
Comment (tkrol88) [inline moler/events/lineevent.py]: set WrongUsage

**Pull request merged**